lib/src/main/java/io/ably/lib/transport/ConnectionManager.java

```diff
@@ -271,6 +272,7 @@ public void run() {
 	 ***************************************/
 
 	public void onMessage(ProtocolMessage message) throws AblyException {
+		Log.v(TAG, "onMessage(): " + new String(ProtocolSerializer.writeJSON(message)));
```

Comment:
I don't like the way that this serialises every message even though the log level doesn't require it to be emitted. I think it has to be done conditionally on the log level, or the log API has to take a format string and arguments so expansion is under control of the logger.